### ISSUE TITLE
rethink (locality?) attribute picks

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#8057]
Help us understand your request (check below):

- [X] new feature or otherwise a good idea to make Arctos even better
- [X] not melt Arctos
- [X] not melt browsers
- [X] not melt users

**Describe what you're trying to do** 

A couple kinds of locality attributes (and others?) seem to have no limit to the number of items from which a user must pick. Currently all categorical attributes use an HTML ``select``, which works well for ~a dozen options. Along with bad UI, the volume of some things is becoming sufficient to melt browsers and begin to threaten Arctos (certainly not in isolation, but nothing happens in isolation).

The easy solution is to send all locality attributes through some other pick mechanism (and agent-like popup or perhaps a search-with-type or etc.), but that would be awkward for things that do and will have that ~dozen values.

Possibly a 'use to select' control could be added to the code table, I'm not sure what that would look like from a programming or user standpoint. (Arctos users seem to value consistency highly, possibly over "good" UI.)

Is there some single alternative that could work for all categorical locality attributes? (Could/should it work for all attributes of all kinds?) Do we need (or are we willing to tolerate!) multiple UIs? Anyone know of any nice models we might follow? 

## HELP!

Calling this a sort of low-level wildfire threat; the table that brought this to my attention (https://arctos.database.museum/info/ctDocumentation.cfm?table=ctquad) has ~3K (and growing rapidly) items, but only ~500K of data. So terrible UI, possibly dangerous to low-spec systems, but probably not much of a system threat - yet.

**Comment by campmlc** [comment]
I don't have a ready answer, other than that seems a good idea, and I do think whatever we do should " work for all attributes of all kinds".

**Comment by dustymc** [comment]
>work for all attributes of all kinds

I think this means that you would prefer quad (~3000 values) and sex (~2 values) to function identically, whatever form that might take.

**Comment by mkoo** [comment]
ah, you already started this one! spitballing....
how about https://jqueryui.com/autocomplete/ (or its equivalent if jquery is not the library of choice)
basically a means so we dont have to scroll 10 miles for the all the topoquads. FYI USGS has over 64k named topoquads!

And I'll add this here too: https://www.sciencebase.gov/catalog/item/60afb6bad34e4043c85648ab
but there has to be a api for the names as there is for the tiled versions for mapping (or maybe you can just use those?)

Otherwise you could go with the popup window search and select option but not a fan for locality attributes.

**Comment by dustymc** [comment]
>https://jqueryui.com/autocomplete/ 

Yes, that could work, it's used in record search (and etc.), but
1. It's old and unsupported (still awesome so whatever...), and
2. https://github.com/ArctosDB/arctos/discussions/8053 - if it's not good for discovery (I think it is!) then it's surely not good for assertion??

>dont have to scroll 10 miles

Nor should your browser run out of memory and eat itself at Mile Seven - autocomplete sends the data in small batches.

>mapping

I think it's a different thing and we need (and can support, there's no conflict) both

* curators say "this is bla quad" because it's in the jar which says 'bla quad', not because any percentage of the locality actually occurs within bla quad. Someone fixing the locality oopsie isn't going to go move a rat to the notbla quad jar.
* researchers might want spatial query for sciencey reasons, and will want to include the things that are from (maybe >75% "from") bla quad but stored in the notbla jar, and also the things from collections which don't assert quads at all.

And that's my take on all of our surprisingly-painful locality lessons over the last few years: There's a ***hard*** line between administrative data and "actual" data; Arctos fortunately and uniquely has the structure to honor that, we don't have to mix up "state according to the collection" (one value, no metadata) and "state according to GIS" (perhaps very complicated for various reasons) - we can easily accommodate both (and curators don't need to know about the latter, except when they put their 'researcher' hats on...).

>popup window search and select

Yea, I think that'd have to be the default, but I don't think it can be made consistent - 19 clicks instead of typing 'f' to assert sex would (rightly, for once!) get the torch-and-pitchfork bearing mobs stomping around on my lawn...
